src/rpc/blockchain.cpp

```diff
@@ -2746,7 +2746,15 @@ static RPCHelpMan loadtxoutset()
     }
 
     SnapshotMetadata metadata;
-    afile >> metadata;
+    try {
+        // Read data from the file
+        afile >> metadata;
+    } catch (const std::exception& e) {
+        // Handle any exception, which includes reaching the end of the file e.g. file size < sizeOf(metadata)
+        throw JSONRPCError(RPC_INTERNAL_ERROR, strprintf("Unable to load UTXO snapshot, "
+            "couldn't read snapshot metadata from file %s.\nThis error is usually "
+            "due to the file size being too small.\n%s", path.utf8string(), e.what()));
```

Comment:
I don't to see how this part of the message would be helpful to users. "file size being too small" does not tell me anything. Would be better to say that the file is corrupt and/or was crafted in an incompatible format. Also, no need to add the internal exception message to the user facing error.

Reply:
Makes sense, thanks!